### PR TITLE
Cleanup inferno plugin (functionality should be the same)

### PR DESCRIPTION
**Pull request by Lucwousin**
Pro tip: Hide whitespace changes

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoWaveMappings.java [hunk 1]
```diff
@@ -26,121 +26,138 @@
...
+
+		panelComponent.getChildren()
+			.add(TitleComponent.builder()
+				.text(header)
+				.color(titleColor)
+				.build()
+			);
+
+
+		for (int i = 1; i < monsters.length; i++)
```
Comment (Kyleeld) [inline runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoWaveMappings.java]: should be for (int i = 0; i < monsters.length; i++)

File: runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoWaveMappings.java [hunk 2]
```diff
@@ -26,121 +26,138 @@
...
 		{
-			return new HashMap<Integer, String>()
+			int monsterType = monsters[i];
+			int count = 1;
+
+			for (; i < monsters.length - 1 && monsters[i + 1] != monsterType; i++)
```
Comment (Kyleeld) [inline runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoWaveMappings.java]: should be for (; i < monsters.length - 1 && monsters[i + 1] == monsterType; i++)

File: runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoWaveOverlay.java [hunk 3]
```diff
@@ -1,69 +1,69 @@
...
-				{
-				renderWave("Next Wave (Wave " + plugin.getNextWaveNumber()  + ")", plugin.getCurrentWaveNumber());
-				}
-			return panelComponent.render(graphics);
+			addWaveComponent(
+				panelComponent,
+				"Current Wave (Wave " + plugin.getCurrentWaveNumber() + ")",
+				plugin.getCurrentWaveNumber(),
+				waveHeaderColor,
+				waveTextColor
```
Comment (Kyleeld) [inline runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoWaveOverlay.java]: change to config.getWaveTextColor()

File: runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoWaveOverlay.java [hunk 4]
```diff
@@ -1,69 +1,69 @@
...
-					&& plugin.isNotFinalWave())
-				{
-				renderWave("Next Wave (Wave " + plugin.getNextWaveNumber()  + ")", plugin.getCurrentWaveNumber());
-				}
-			return panelComponent.render(graphics);
+			addWaveComponent(
+				panelComponent,
+				"Current Wave (Wave " + plugin.getCurrentWaveNumber() + ")",
+				plugin.getCurrentWaveNumber(),
+				waveHeaderColor,
```
Comment (Kyleeld) [inline runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoWaveOverlay.java]: change to config.getWaveOverlayHeaderColor(),

File: runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoWaveOverlay.java [hunk 5]
```diff
@@ -1,69 +1,69 @@
...
-					{
-					continue;
-					}
-				panelComponent.getChildren().add(TitleComponent.builder().text(quantity + "x " + plugin.getMonster().get(monsterID)).color(config.getWaveTextColor()).build());
-				}
+			addWaveComponent(
+				panelComponent,
+				"Next Wave (Wave " + plugin.getNextWaveNumber()  + ")",
+				plugin.getNextWaveNumber(),
+				waveHeaderColor,
```
Comment (Kyleeld) [inline runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoWaveOverlay.java]: change to config.getWaveOverlayHeaderColor(),

File: runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoWaveOverlay.java [hunk 6]
```diff
@@ -1,69 +1,69 @@
...
+				waveTextColor
```
Comment (Kyleeld) [inline runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoWaveOverlay.java]: change to config.getWaveTextColor()

File: runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoPlugin.java [hunk 7]
```diff
@@ -1,439 +1,469 @@
...
+		if (isInInferno())
+		{
+			overlayManager.add(infernoOverlay);
+			overlayManager.add(infernoInfobox);
+			overlayManager.add(nibblerOverlay);
+
+			if (config.waveDisplay() != InfernoWaveDisplayMode.NONE)
+			{
+				overlayManager.add(waveOverlay);
+			}
```
Comment (Kyleeld) [inline runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoPlugin.java]: delete 
			if (config.waveDisplay() != InfernoWaveDisplayMode.NONE)
			{
				overlayManager.add(waveOverlay);
			}

add overlayManager.add(waveOverlay); to startup

**Comment by Kyleeld** [comment]
suggested fixes fix:

wave overlay colouring issue

waves not building correctly

outstanding issues: NPE on startup

File: runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoWaveOverlay.java (hunk 3 above)
Comment (Lucwousin) [reply runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoWaveOverlay.java]: no, wanna get away from using config.something in overlays cause it's way slower

File: runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoPlugin.java (hunk 7 above)
Comment (Lucwousin) [reply runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoPlugin.java]: No, wanna go away from having overlays statically on so their render methods don't get called

**Pull request merged**